src/sys/windows/selector.rs

```diff
@@ -438,18 +438,8 @@ impl SelectorInner {
     pub fn select(&self, events: &mut Events, timeout: Option<Duration>) -> io::Result<()> {
         events.clear();
 
-        if timeout.is_none() {
-            loop {
-                let len = self.select2(&mut events.statuses, &mut events.events, None)?;
-                if len == 0 {
-                    continue;
-                }
-                return Ok(());
-            }
-        } else {
-            self.select2(&mut events.statuses, &mut events.events, timeout)?;
-            return Ok(());
-        }
+        self.select2(&mut events.statuses, &mut events.events, timeout)?;
```

Comment:
There is a reason why the code was a little complicated: https://github.com/tokio-rs/mio/issues/1106, https://github.com/tokio-rs/mio/commit/bbd82672ab5adab7cd68e765810d1d1809b90c46. Are you sure this still works?

Reply:
Spurious wakeups are legal. Either way, I am reverting.